src/MustacheTemplateController.js

```diff
@@ -51,19 +52,31 @@ module.exports = class MustacheTemplateController extends BaseTemplateController
     return s;
   }
 
+  _renderTemplateItem(item, view, engine = 'mustache') {
+    if (engine === 'handlebars') {
+      let template = Handlebars.compile(item);
+      return template(view);
+    }
+    return Mustache.render(item, view);
+  }
+
   async processTemplate(templates, view) {
     let customTags = objectPath.get(this.data, ['object', 'spec', 'custom-tags']);
+    let templateEngine = objectPath.get(this.data, 'object.spec.templateEngine', 'mustache').toLowerCase();
+    this.log.info(`MustacheTemplateController: Using ${templateEngine} template engine`);
+
+    if (templateEngine === 'handlebars') { Handlebars.registerHelper(HandlebarHelpers); }
+
     let templatesArr = await this._stringifyTemplates(templates);
     let tempTags = Mustache.tags;
     if (customTags) {
       Mustache.tags = customTags;
     }
```

Comment:
should throw an error if user tries to use custom tags with handlebars
```suggestion
  let templatesArr = await this._stringifyTemplates(templates);
  let tempTags = Mustache.tags;

  if (templateEngine === 'handlebars') {
    Handlebars.registerHelper(HandlebarHelpers);
  }
  if (customTags && templateEngine === 'handlebars') {
    return Promise.reject({ statusCode: 500, message: 'unable to set customTags when using handlebars' });
  } else if (customTags && templateEngine === 'mustache') {
    Mustache.tags = customTags;
  }
```

Reply:
implemented